course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt

```diff
@@ -611,6 +612,404 @@ fun VideoSubtitles(
+                textAlign = TextAlign.Center
+            )
+        }
+    }
+}
+
+@Composable
+fun CourseUnitToolbar(
+    title: String,
+    onBackClick: () -> Unit
+) {
+    OpenEdXTheme {
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .displayCutoutForLandscape()
+                .zIndex(1f)
+                .statusBarsPadding()
+        ) {
+            BackBtn { onBackClick() }
+            Text(
+                modifier = Modifier
+                    .padding(horizontal = 56.dp)
+                    .align(Alignment.Center),
+                text = title,
+                color = MaterialTheme.appColors.textPrimary,
+                style = MaterialTheme.appTypography.titleSmall,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis
+            )
+        }
+    }
+}
+
+@Composable
+fun UnitSubSectionsTitle(
+    subSectionName: String,
+    subSectionsCount: Int,
+    blockListShowed: Boolean?,
+    onBlockClick: () -> Unit
+) {
+    val textStyle = MaterialTheme.appTypography.titleMedium
+    val hasSections = subSectionsCount > 0
+    var rowModifier = Modifier
+        .fillMaxWidth()
+        .padding(
+            horizontal = 16.dp,
+            vertical = 8.dp
+        )
+        .displayCutoutForLandscape()
+    if (hasSections) {
+        rowModifier = rowModifier.noRippleClickable { onBlockClick() }
+    }
+
+    Row(
+        modifier = rowModifier,
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(4.dp)
+    ) {
+        Text(
+            modifier = Modifier
+                .weight(1f),
+            text = subSectionName,
+            color = MaterialTheme.appColors.textPrimary,
+            style = textStyle,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            textAlign = TextAlign.Start
+        )
+
+        if (hasSections) {
+            Icon(
+                modifier = Modifier.rotate(if (blockListShowed == true) 180f else 0f),
+                painter = painterResource(id = R.drawable.ic_course_arrow_down),
+                contentDescription = null,
+                tint = MaterialTheme.appColors.textPrimary
+            )
+        }
+    }
+}
+
+@Composable
+fun UnitSubSectionsList(
+    sectionsBlocks: List<Block>,
+    selectedSection: Int = 0,
+    onSectionClick: (index: Int, block: Block) -> Unit
+) {
+    Card(
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight(align = Alignment.Top)
+            .displayCutoutForLandscape(),
+        border = BorderStroke(1.dp, MaterialTheme.colors.onSurface.copy(alpha = 0.12f))
+    ) {
+        LazyColumn(Modifier.fillMaxWidth()) {
+            itemsIndexed(sectionsBlocks) { index, block ->
+                Column {
+                    if (block.isGated() && (index == 0 || !sectionsBlocks[index - 1].isGated())) {
```

Comment:
in this part of the code we need to show a banner that will tell the user that this unit is gated. like here:

<img width="150"  src="https://github.com/openedx/openedx-app-android/assets/5153976/46d496b7-7218-40b9-b56b-1dd49d6a340b">

I thought that all gated units will be at the bottom of the list (didn't see other options). But anyway, using my logic this banner will be showed at the top of any gated unit in case that the previous unit is not gated.

thoughts or some additional information?)

Reply:
done